Incorrect Assignment Id format throws exception, but error is not shown to user

![image](https://github.com/user-attachments/assets/cf99d8ab-8a4c-4d85-97ac-7e73d120013f)

Incorrect Assignment Id format throws exception, but error is not shown to user.

Exception thrown:
`Caused by: java.lang.IllegalArgumentException: Assignment ID should only contain numeric characters`

--- a/src/main/java/seedu/address/logic/parser/ParserUtil.java
+++ b/src/main/java/seedu/address/logic/parser/ParserUtil.java
@@ -208,8 +208,8 @@ public class ParserUtil {
     public static AssignmentId parseAssignmentId(String assignmentId) throws ParseException {
         requireNonNull(assignmentId);
         String trimmedAssignmentId = assignmentId.trim();
-        if (!ProjectId.isValidId(trimmedAssignmentId)) {
-            throw new ParseException(ProjectId.MESSAGE_CONSTRAINTS);
+        if (!AssignmentId.isValidAssignmentId(trimmedAssignmentId)) {
+            throw new ParseException(AssignmentId.MESSAGE_CONSTRAINTS);
         }
         return new AssignmentId(trimmedAssignmentId);
     }

--- a/src/test/java/seedu/address/logic/parser/ParserUtilTest.java
+++ b/src/test/java/seedu/address/logic/parser/ParserUtilTest.java
@@ -315,7 +315,7 @@ public class ParserUtilTest {
     }
 
     @Test
-    public void parseAssignmentId_validValue_returnTrimedAssignmentId() throws Exception {
+    public void parseAssignmentId_validValue_returnTrimmedAssignmentId() throws Exception {
         AssignmentId expectedAssignmentId = new AssignmentId(VALID_ASSIGNMENT_ID);
         assertEquals(expectedAssignmentId, ParserUtil.parseAssignmentId(VALID_ASSIGNMENT_ID));
     }

--- a/src/test/java/seedu/address/logic/parser/ParserUtilTest.java
+++ b/src/test/java/seedu/address/logic/parser/ParserUtilTest.java
@@ -51,6 +51,7 @@ public class ParserUtilTest {
     private static final String VALID_PROJECT_NAME = "Project Alpha";
 
     private static final String VALID_ASSIGNMENT_ID = "1";
+    private static final String INVALID_ASSIGNMENT_ID = "A";
 
     private static final String WHITESPACE = " \t\r\n";
 
@@ -318,5 +319,10 @@ public class ParserUtilTest {
     public void parseAssignmentId_validValue_returnTrimmedAssignmentId() throws Exception {
         AssignmentId expectedAssignmentId = new AssignmentId(VALID_ASSIGNMENT_ID);
         assertEquals(expectedAssignmentId, ParserUtil.parseAssignmentId(VALID_ASSIGNMENT_ID));
+    }
+
+    @Test
+    public void parseAssignmentId_collectionWithInvalidSkills_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseAssignmentId(INVALID_ASSIGNMENT_ID));
     }
 }

--- a/src/test/java/seedu/address/logic/parser/ParserUtilTest.java
+++ b/src/test/java/seedu/address/logic/parser/ParserUtilTest.java
@@ -14,6 +14,7 @@ import java.util.Set;
 import org.junit.jupiter.api.Test;
 
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.assignment.AssignmentId;
 import seedu.address.model.person.Address;
 import seedu.address.model.person.Email;
 import seedu.address.model.person.EmployeeId;
@@ -48,6 +49,8 @@ public class ParserUtilTest {
 
     private static final String VALID_PROJECT_ID = "A0276123K";
     private static final String VALID_PROJECT_NAME = "Project Alpha";
+
+    private static final String VALID_ASSIGNMENT_ID = "1";
 
     private static final String WHITESPACE = " \t\r\n";
 
@@ -309,5 +312,11 @@ public class ParserUtilTest {
         String nameWithWhitespace = WHITESPACE + VALID_PROJECT_NAME + WHITESPACE;
         ProjectName expectedProjectName = new ProjectName(VALID_PROJECT_NAME);
         assertEquals(expectedProjectName, ParserUtil.parseProjectName(nameWithWhitespace));
+    }
+
+    @Test
+    public void parseAssignmentId_validValue_returnTrimedAssignmentId() throws Exception {
+        AssignmentId expectedAssignmentId = new AssignmentId(VALID_ASSIGNMENT_ID);
+        assertEquals(expectedAssignmentId, ParserUtil.parseAssignmentId(VALID_ASSIGNMENT_ID));
     }
 }

